src/Native/System.Native/pal_exec.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "../config.h"
+#include "pal_exec.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+enum
+{
+    READ_END_OF_PIPE = 0,
+    WRITE_END_OF_PIPE = 1,
+};
+    
+inline static void CloseIfOpen(int fd)
+{
+    if (fd >= 0)
+        close(fd);
+}
+
+extern "C"
```

Comment:
Lest anyone think it'd be a good idea to move this function back to managed code,I think a small comment here would be helpful, indicating why we can't do the fork/exec combination from managed.  Essentially, forking only clones the current thread, not all threads in the process, which makes it very unsafe given things like GC, finalization, etc.  We could, for example, end up in deadlock situations by trying to run managed code in the child process between the fork and the exec.